lib/cloud-metadata/aws.js

```diff
@@ -118,50 +95,36 @@ function getMetadataAwsV2 (host, port, socketTimeoutMs, httpTimeout, protocol, c
         finalData.push(data)
       })
 
-      res.on('end', function (data) {
-        try {
-          const headers = {
-            'X-aws-ec2-metadata-token': finalData.join('')
-          }
-          getMetadataAws(
-            host,
-            port,
-            headers,
-            socketTimeoutMs,
-            httpTimeout,
-            protocol,
-            cb
-          )
-        } catch (e) {
-          cb(e)
+      res.on('end', function () {
+        const token = sanitizeHttpHeaderValue(finalData.join(''))
```

Comment:
This sanitization is just abundance-of-caution, rather than a particular concern of a garbled-yet-HTTP-200-response?

Reply:
An abundance of caution and something that came up in tests -- a 200 response but with an HTML message-body  that has newlines will produce an invalid header, _and_ the invalid header is a case where Node.js will throw an exception.  By sanitizing the header and avoiding the possibility of this thrown exception we avoid that additional try/catch further above.

Reply:
> something that came up in tests

Good to know, thanks. That seems surprising that the token call would sometimes yield newlines and sometimes not.